transcom/mymove
# INT- B-21676 exclude nsra15 duty locations from search

## [B-21676](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1031218)

## Summary

This PR addresses issue related to search AK duty locations that are classified as Level V(RateArea: NSRA15). Approximately 200+ AK duty locations have been classified as NSRA15 rate area. This PR updates the duty location search to exclude NSRA15 locations from search results. 

[NSRA15-duty-locations-data-1731515178779.csv](https://github.com/user-attachments/files/17838343/NSRA15-duty-locations-data-1731515178779.csv)

 This PR contains migration file to flag specific re_oconus_rate_areas records as inactive(active=false). This flag serves as the basis for the exclusion mechanism. In addition, migration file also contains INSERTs to correct 18 duty locations not having any oconus rateAea mappings(provided by Beth Grohmann).

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

We want to verify each duty location identified as NSRA15 affiliated is excluded from search result by its formal name, alternative names and postal code. Note: duty locations are searchable by complete/partial name and postal code.

1. Login as customer and start process of onboarding a move.
2. Using the NSRA15 file attached to this PR, search names/postal codes and verify duty locations are not returned in search results. Make sure to also do a search only using its postal code.
3. Search using known alternative names. Verify the alternative name does not return results containing its known location.
ex) 
"Brevig Msn, AK 99785" or "Brevig Msn" or  "99785" => Brevig Mission, AK 99785
"Arctic Vlg, AK 99722" or ""Arctic Vlg" or 99722" =>  Arctic Village, AK 99722
"Anaktuvuk, AK 99721" or "Anaktuvuk" or 99721" => Anaktuvuk Pass, AK 99721


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
